### PR TITLE
Documentation: Add markdown syntax highlighting to code blocks

### PR DESCRIPTION
**Pull request by spen**
While looking over the mixins / HoCs usage I noticed that one of the files could have benefited from syntax highlighting in the `readme` markdown. I went on a bit of a rampage and looked for any code blocks that were missing syntax highlighting and added in the syntax that made sense to me...

The _'rules'_ that I used to determine which syntax to use (with the option of `js`, `jsx` and `es6`) were something like this:
`js` - mostly just for basic objects on their own, i.e:
```js
{
    something: 'objecty',
}
```

`jsx` - this ones quite obvious -use if the block includes `<Jsx />` 

`es6` pretty much anything else that _could_ include es6 syntax... even if it's currently < es6

I've also refrained from doing anything other than adding the ` ```syntax ` to the code blocks in these docs - as tempting as it was to go on an even bigger refactoring rampage!

'Testing' should be very easy here, please just take a look at each `readme` and make sure the syntax highlighting makes sense. If you think any code block should be tagged with a different syntax just holla and I'll make a change!

Also, regarding the refactoring of these `readme`s - I'm quite keen to update any that are not on their way out (mixins hopefully?) - I'll open a separate issue and include the list that I made of the files that could use a face-lift :D

In case you're not convinced...

```
<UpgradeNudgeExpanded
    plan={ PLAN_BUSINESS }
    title={ this.translate( 'Upgrade to a Business Plan and Enable Advanced SEO' ) }
    subtitle={ this.translate( 'By upgrading to a Business Plan you\'ll enable advanced SEO features on your site.' ) }
    highlightedFeature={ FEATURE_ADVANCED_SEO }
    benefits={ [
        this.translate( "Preview your site's posts and pages as they will appear when shared on Facebook, Twitter and the WordPress.com Reader." ),
        this.translate( 'Allow you to control how page titles will appear on Google search results, or when shared on social networks.' ),
        this.translate( 'Modify front page meta data in order to customize how your site appears to search engines.' )
    ] }
/>
```

becomes:

```jsx
<UpgradeNudgeExpanded
    plan={ PLAN_BUSINESS }
    title={ this.translate( 'Upgrade to a Business Plan and Enable Advanced SEO' ) }
    subtitle={ this.translate( 'By upgrading to a Business Plan you\'ll enable advanced SEO features on your site.' ) }
    highlightedFeature={ FEATURE_ADVANCED_SEO }
    benefits={ [
        this.translate( "Preview your site's posts and pages as they will appear when shared on Facebook, Twitter and the WordPress.com Reader." ),
        this.translate( 'Allow you to control how page titles will appear on Google search results, or when shared on social networks.' ),
        this.translate( 'Modify front page meta data in order to customize how your site appears to search engines.' )
    ] }
/>
```

So much prettier! 💅

TIA!

### CODE REVIEW

**Comment by alisterscott** [comment]
Thanks for your contribution @spen 🙏

The changes look good to me, but should I see this highlighting when I open devdocs? For example, I compare this devdocs page [here](https://wpcalypso.wordpress.com/devdocs/client/components/data/query-active-theme/README.md), with this same page on calypso.localhost with these changes and it looks the same - should it show syntax highlighting here or does it serve another purpose?

Thanks

**Comment by spen** [comment]
NP @alisterscott 
The syntax highlighting shows up on GitHub, which is where I tend to look when reading the readmes. 
I've not looked too deeply in to devDocs but my guess is that the markdown parser doesn't handle the syntax highlighting (could be room for improvement if not :D ).

File: client/components/empty-content/README.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ The component also supports a secondary action. This should be used sparingly.
```
Comment (spen) [inline client/components/empty-content/README.md]: The 'usage' code block above could do with being updated from ` ```js ` to ` ```jsx `

File: client/lib/plugins/README.md [hunk 2]
```diff
@@ -180,7 +180,7 @@ Toggle AutoUpdates for a plugin on a site.
 
 ####Example Component Code:
 
-```
+```es6
```
Comment (spen) [inline client/lib/plugins/README.md]: I overlooked the jsx here, update coming

File: client/lib/plugins/wporg-data/README.md [hunk 3]
```diff
@@ -105,7 +105,7 @@ Triggers api call to fetch the plugin data and update the store.
 
 ####Example Component Code:
 
-```
+```es6
```
Comment (spen) [inline client/lib/plugins/wporg-data/README.md]: jsx

**Comment by alisterscott** [comment]
Verified the difference when viewing files via Github:

**Before:**

before

**After:**

after

**Comment by spen** [comment]
Thanks for checking over @alisterscott ! I just have a few areas that I'd like to improve as mentioned in my self-review :) They'll be up in a jiffy!

**Comment by spen** [comment]
Annnnnd done - They should all be good to go now! :D

**Comment by spen** [comment]
Thanks @alisterscott :)

**Pull request merged**